src/tools/rbd/action/Device.cc

```diff
@@ -189,6 +205,43 @@ int execute_unmap(const po::variables_map &vm,
   return (*get_device_operations(vm)->execute_unmap)(vm, ceph_global_init_args);
 }
 
+void get_attach_arguments(po::options_description *positional,
+                          po::options_description *options) {
+  add_device_type_option(options);
+  at::add_image_or_snap_spec_options(positional, options,
+                                     at::ARGUMENT_MODIFIER_NONE);
+  options->add_options()
+    ("device", po::value<std::string>()->required(), "specify device path")
```

Comment:
This only mark it as required in the output, you still need to check if it is specified and if not bail.

Reply:
oops! I remember I tested this and the command failed when the device option is absent, maybe that was from the rbd-nbd backend. Thanks!